### PR TITLE
Travis integration

### PR DESCRIPTION
**Pull request by HieronymusLex**
* add github releaser
* add pylint
* fix linter errors

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+before_script:
+  - python --version
+  - pip install -r requirements.txt
+
+script:
+  - find lambdas/ -iname "*.py" | xargs pylint --rcfile .pylintrc
+  - AWS_DEFAULT_REGION=eu-west-1 python -m unittest discover tests
+
+before_deploy:
+  - export TRAVIS_TAG=${TRAVIS_TAG:-v$MAJOR_VERSION.$TRAVIS_BUILD_NUMBER}
```
Comment (HieronymusLex) [inline .travis.yml]: @stilvoid I think `TRAVIS_BUILD_NUMBER` may lead to gaps in the release numbers? semver probably isn't appropriate here, so perhaps it's more appropriate to get the latest tag, and increment by 1?

File: git-tag.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (HieronymusLex) [inline git-tag.sh]: @stilvoid this versioning stuff is for this repo - it'll have no impact on cfn-publish which versioning-strategy agnostic

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+language: python
+
+python: 
+  - "3.7"
+
+env:
+- MAJOR_VERSION="v0"
```
Comment (stilvoid) [inline .travis.yml]: Missing indent

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  - NEW_TAG="$(./git-tag.sh)"
+  - export TRAVIS_TAG=${TRAVIS_TAG:-$NEW_TAG}
+  - echo "Deploying version $TRAVIS_TAG"
+  - git tag $TRAVIS_TAG
+  - git archive HEAD -o $TRAVIS_TAG.zip
+
+deploy:
+- provider: releases
+  api_key: 
+    secure: iCddiGenu0FqAmlSX8ERfIzkDdeai44N6PNJ4eowrioCOMniab/nqzkkGlOP30+dEfNe9TAykExJzkzI+nX2sAV/3TKr1URcmd0bzbdGh/t93sixoVfeo+pjfi9oUoND9NsPOcwS1djHesRZjRRaEfpsYDlrp15jNLtZLpuyZQRI8KAtzcoDRFYgVKorKxzFrBpK+T/6qbFWiE2X4fU1oZWURi0FYAMRW8DxOzWF1kPuP9jjzcnLU8nH5ooyx9QtTLutEMg/p9p0w+4+Bz3GmnvQ7FvQlZIJsZib9DFGvxfHss9tWgGxLU99tQ2WeTXQ0IfJstKjHPZw0wH5zj7lNfJc5cq+xs6xHEvr/jdrnBgpJKk8TKw2+heGa3Io+NH3hoxyshnSE9fiU9L+xnnaqxcIfIjnSzz5ri9BcDSeMxlkNm3ZxCWkXzupL38AjkWJ4c76B5i58J8QhKpLjGWmewWWz2GG227Jrq5Q5EzmXpOBX4gY3gtkjE+JMBEQ5K238+7cS+Q5g8hjOSf5wkzMLZKSZ5XRYFC0voq8P+YHmpBUscEfJQtD2UJQzHbIgZh262YZ4FOR4eDjcO3XMBbMsu1i+q1dFstPaxcfGJTROOEU5sDG7KcRmtB6KdHTUoJGFw1evfxuMP2nSKBQukJN5fxwYuL4zaZ4y/srtPAZDzc=
```
Comment (stilvoid) [inline .travis.yml]: Not secure now it's in a commit :rofl:

File: git-tag.sh (hunk 2 above)
Comment (stilvoid) [reply git-tag.sh]: So this repo will increment its own version on deployment?

If it's not semver, why a segmented version number? Could just go v1 v2 v3, etc.

File: lambdas/index.py [hunk 5]
```diff
@@ -17,16 +20,16 @@ def handler(event, context):
...
     for record in event["Records"]:
-        try:    
+        try:
             bucket = record["s3"]["bucket"]["name"]
             object_key = record["s3"]["object"]["key"]
             build_data = get_build_data(object_key)
             start_build(build_data["version"], "{bucket}/{key}".format(bucket=bucket, key=build_data["object_key"]))
         except IndexError as e:
-            logger.error("Unable to determine artefact project and version from object key: {}. Uploads should be in the format s3://BUCKET/PROJECT/VERSION/PROJECT.zip".format(object_key))
+            logger.error("Unable to determine artefact project and version from object key: %s. Uploads should be in the format s3://BUCKET/PROJECT/VERSION/PROJECT.zip", object_key)
```
Comment (stilvoid) [inline lambdas/index.py]: Why switching to old style string interp?

File: lambdas/index.py [hunk 6]
```diff
@@ -17,16 +20,16 @@ def handler(event, context):
...
             raise e
         except ValueError as e:
-            logger.error("Invalid object key {}:".format(record["s3"]["object"]["key"]))
+            logger.error("Invalid object key: %s", record["s3"]["object"]["key"])
```
Comment (stilvoid) [inline lambdas/index.py]: And again...

File: lambdas/index.py (hunk 5 above)
Comment (HieronymusLex) [reply lambdas/index.py]: pylint pointed this out. when using a logger, using this method does interpolation lazily if the message is going to be displayed https://docs.python.org/3/howto/logging.html#optimization

tbh in this situation there the performance difference is negligible and it'd be rare that log level would be set such that error won't be printed, however apparently it is technically the correct way forward. it's not actually quite the old style of `"blah blah %s" % some_arg`, the interpolation vars are supplied as args to the logger method

File: git-tag.sh (hunk 2 above)
Comment (HieronymusLex) [reply git-tag.sh]: v good point - will switch to that

**Pull request merged**